gapis/api/vulkan/api/descriptor.api

```diff
@@ -256,12 +268,18 @@ cmd VkResult vkAllocateDescriptorSets(
   if pAllocateInfo == null { vkErrorNullPointer("VkDescriptorSetAllocateInfo") }
   info := pAllocateInfo[0]
   // handle pNext
+  variableDescriptorCounts := MutableU32S()
   if info.pNext != null {
     numPNext := numberOfPNext(info.pNext)
     next := MutableVoidPtr(as!void*(info.pNext))
     for i in (0 .. numPNext) {
       sType := as!const VkStructureType*(next.Ptr)[0]
       switch (sType) {
+        case VK_STRUCTURE_TYPE_DESCRIPTOR_SET_VARIABLE_DESCRIPTOR_COUNT_ALLOCATE_INFO: {
+          ext := as!VkDescriptorSetVariableDescriptorCountAllocateInfo*(next.Ptr)[0]
+          variableDescriptorCounts.Val = ext.pDescriptorCounts[0:ext.descriptorSetCount]
+          read(ext.pDescriptorCounts[0:ext.descriptorSetCount])
```

Comment:
I don't know what you mean with first or second? do you mean whether the read should be before the assignment to `variableDescriptorCounts.Val`? if so, it doesn't matter, since the first one is just a pointer assignment. In reality, the read is going to be unnecessary, because we'll dereference `variableDescriptorCounts.Val` below, but it makes sense here, since relying on the derefs below may leave holes in the array.

Reply:
In my testing I was missing memory observations without the explicit read, I suspect because the `variableDescriptorCounts.Val` slice isn't accessed until after `fence`

Reply:
hmm... I guess the automatic fence placement misses this read, probably due to the indirection.